modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -24,15 +24,18 @@ import * as InsertList from './InsertList';
 
 const wrappedElements = [ 'pre' ];
 
-const shouldPasteContentOnly = (fragment: AstNode, parentNode: Element) => {
-  const firstNode = fragment.firstChild;
-
-  const isAFlattenableTag = Arr.contains(wrappedElements, firstNode.name);
-  const isPastingInTheSameTag = firstNode.name === parentNode.tagName.toLowerCase();
-  const lastNode = fragment.lastChild.attr('data-mce-type') === 'bookmark' ? fragment.lastChild.prev : fragment.lastChild;
-  const isCopingOnlyOneTag = firstNode === lastNode;
-
-  return isCopingOnlyOneTag && isAFlattenableTag && isPastingInTheSameTag;
+const shouldPasteContentOnly = (dom: DOMUtils, fragment: AstNode, parentNode: Element, editor: Editor): boolean => {
```

Comment:
Instead of passing the `editor` here, it'd be cleaner and easier to test in isolation if we pass the root element through to this function instead since that's all we're using `editor` for. For example:
```ts
const shouldPasteContentOnly = (dom: DOMUtils, fragment: AstNode, parentNode: Element, root: Node): boolean => {
```

Reply:
Did consider that, will do.